libs/core/src/lib/token/tokenizer.component.ts

```diff
@@ -229,16 +245,39 @@ export class TokenizerComponent implements AfterViewChecked, AfterContentInit, O
         }
     }
 
+    /** @hidden */
+    @HostListener('document: click', ['$event'])
+    clickout(event) {
+        if (this.elementRef().nativeElement.contains(event.target) === false) {
+            this.tokenList.forEach(token => {
+                token.selected = false;
+            });
+        }
+    }
+
     /** @hidden */
     handleKeyDown(event: KeyboardEvent, fromIndex: number): void {
         let newIndex: number;
         const rtl = this._rtlService && this._rtlService.rtl ? this._rtlService.rtl.getValue() : false;
-        if ((event.code === 'ArrowLeft' && !rtl) || (event.code === 'ArrowRight' && rtl)) {
+        if (KeyUtil.isKey(event, 'Space')) {
+            const token = this.tokenList.filter((element, index) => index === fromIndex);
+            this.tokenList.forEach(shadowedToken => {if (shadowedToken !== token[0]) {shadowedToken.selected = false}});
+            token[0].selected = !token[0].selected;
+            event.preventDefault();
```

Comment:
Could you just unselect all tokens from tokenList and then switch current one? I'm not sure about keeping selected non-visible tokens.

Reply:
Not sure I totally understand what you're saying... do you mean we don't need the "if" at line 264?  If I am trying to unselect token[0] and it is selected...  If I unselect every token in the tokenList.forEach, then token[0] will be unselected, and then it will be re-selected after the next line

Reply:
Actually you are right, it won't be possible to unselect. NVM.